pennylane/_qubit_device.py

```diff
@@ -769,20 +773,19 @@ def statistics(
               ``shot_range=[35, 135]``, ``bin_size=100``.
         """
         if observables is not None:
-            if isinstance(observables, QuantumScript):
-                circuit = observables
-                measurements = circuit.measurements
-            else:
-                warnings.warn(
-                    message="Using a list of observables in ``QubitDevice.statistics`` is "
-                    "deprecated. Please use a ``QuantumTape`` instead.",
-                    category=UserWarning,
-                )
-                measurements = observables
-        elif circuit is not None:
+            circuit = observables
```

Comment:
We still have the same problem, if the user inputs both `observables` and `circuit`, `circuit` will get overriden.

I would prefer leaving the code as it is in master and just updating the warning message.

Reply:
Doesn't what's on the rc branch also override circuit if both observables and circuit are specified?

Reply:
From rc branch - looks like it only looks at `circuit` if `observables` was not None, redefines `circuit` based on `observables` if `observables` are passed:

```
        if observables is not None:
            if isinstance(observables, QuantumScript):
                circuit = observables
                measurements = circuit.measurements
            else:
                warnings.warn(
                    message="Using a list of observables in ``QubitDevice.statistics`` is "
                    "deprecated. Please use a ``QuantumTape`` instead.",
                    category=UserWarning,
                )
                measurements = observables
        elif circuit is not None:
            measurements = circuit.measurements
        else:
            raise ValueError("Please provide a circuit into the statistics method.")
```

Reply:
Yes. Given that `observables` is at the first argument, when it is not `None` it might be for 3 different reasons:
1. `self.statistics(tape)`: user passes a tape -> override `circuit = observables`
2. `self.statistics([])`: user passes a list -> raise warning
3. `self.statistics(observables=[])`: user uses kwarg -> raise warning

Reply:
If `observables` is `None` it means that the user is doing: `self.statistics(circuit=tape)`, thus we don't need to raise any warning.

Reply:
Updated now!